pymedphys/tests/dicom/rtplan/test_leaf_jaw_pos.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2020 Simon Biggs
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from pymedphys._dicom import create, rtplan
+
+
+def test_get_leaf_jaw_position_of_type():
+
+    rt_beam_limiting_device_type = "ASYMY"
+    leaf_jaw_positions = [0, 0]
+
+    expected_positions = [leaf_jaw_positions] * 4
+
+    control_point_sequence = create.dicom_dataset_from_dict(
+        {
+            "ControlPointSequence": [
+                {
+                    "BeamLimitingDevicePositionSequence": [
+                        {
+                            "RTBeamLimitingDeviceType": rt_beam_limiting_device_type,
+                            "LeafJawPositions": leaf_jaw_positions,
+                        },
+                        {
+                            "RTBeamLimitingDeviceType": "Something Else!",
+                            "LeafJawPositions": [1, 1],
+                        },
+                    ]
+                },
+                {},
+                {},
+                {},
+            ]
+        }
+    ).ControlPointSequence
+
+    beam_limiting_device_position_sequences = rtplan.get_cp_attribute_leaning_on_prior(
```

Comment:
I'd need to go dig through to see exactly what get_cp_attribute_leaning_on_prior offers as it's contract, but if you are passing in a sequence with a nearly arbitrary number of items, I'm surprised you don't have to pass in the control point index (not the item index, but the control point index).  Maybe it defaults to zero, which would be reasonable.

Reply:
I've seen that often in DICOM files if a leaf and/or jaw position doesn't change in one control point to the next then it actually just uses the result of the previous control point. As in, if the position isn't defined it is presumed to not move with respect to the previous control point.

https://github.com/pymedphys/pymedphys/blob/2cf5dacc50fd841c9a9ab07dd97a06fcb748b80e/pymedphys/_dicom/rtplan/core.py#L162-L178

I believe this may also shed light on https://github.com/pymedphys/pymedphys/pull/1048#discussion_r487453031

Reply:
> (not the item index, but the control point index)

I guess I had hoped... potentially inappropriately, that the order of the control point sequence is consistent. I'm not sure, is the control point order within `ControlPointSequence` strictly increasing?

Reply:
Yes, there is a Run Length Encoding approach specified in DICOM.
But the standard also says that if the value *never* changes, you can use the RLE approach, and just put it in with the control point with Control Point Index == 0.  
And if it does *ever* change, you need to put in a value in each control point (even if it is all duplicates except for one).

What the code tells me is that the function returns a list of values, with the list being of length "Number of Control Points", and if the value is itself a list, you are going to receive a list of lists.
I encourage you to put in docstring that says something on the order of
"""returns a list of values, one list item per control point, unpacking any run length encoding
"""

Also, will the values returned be of the conceptual type most closely represented in Python, e.g.
a floating point value if the DICOM Element has a VR of DS, or will it be literally of the DICOM VR (so DS comes back as a string)?
If the attribute is a Sequence (VR is SQ), is the return a list of dataset or a list of iterables of the Sequence Item?

I think it will be very useful for it to be clear what the function returns for a variety of attributes.
While I'm not a fan of changing signatures, the name of the function suggested to me that it was trying to get a single value.  I'm not sure of the python terminology, but you are slicing across a dimension (the attribute axis?) (and getting back something that is of the dimension "dimension of attribute value + 1" (where the 1 comes from the control point/index axis).  For something like a Gantry Angle, the value is a scalar and has dimension 0.
If this was new code, I'd push for a name that more clearly reflects that.

Reply:
> I'm not sure, is the control point order within `ControlPointSequence` strictly increasing?

It's a pretty safe bet.  But it's not strictly guaranteed by the standard that items in a sequence will be ordered monotonically increasing according to the index element that is used to formally identify the sequence item when it is being referenced.  I have not run in to data that came out of order for control points.  But that doesn't mean a bit of paranoia is unjustifiable.